fastlane/spec/plugins_specs/plugin_manager_spec.rb

```diff
@@ -38,13 +39,32 @@
 
       it "returns all fastlane plugins with no fastlane_core" do
         allow(Bundler::SharedHelpers).to receive(:default_gemfile).and_return("./fastlane/spec/fixtures/plugins/Pluginfile1")
+        allow(Bundler::SharedHelpers).to receive(:default_lockfile).and_return(nil)
         expect(plugin_manager.available_plugins).to eq(["fastlane-plugin-xcversion"])
       end
+
+      it "returns all fastlane plugins with no fastlane_core and a gemfile lock" do
+        allow(Bundler::SharedHelpers).to receive(:default_gemfile).and_return("./fastlane/spec/fixtures/plugins/GemfileWithLock")
+        allow(Bundler::SharedHelpers).to receive(:default_lockfile).and_return("./fastlane/spec/fixtures/plugins/GemfileWithLock.lock")
+        expect(plugin_manager.available_plugins).to eq(["fastlane-plugin-appcenter"])
+      end
+
+      # full integration test
+      it "returns all fastlane plugins found in transitive dependencies" do
+        Bundler.with_unbundled_env do
+          Dir.chdir("fastlane/spec/fixtures/plugins/GemfileWithDeps") do
+            `bundle install`
+            output = `bundle exec fastlane lanes`
+            expect(output.index("fastlane-plugin-appcenter")).not_to eq(nil)
```

Comment:
I think this might also work 👇 

```suggestion
            expect(output.include?("fastlane-plugin-appcenter")).to be(true)
```